apps/prairielearn/src/lib/server-jobs-2.ts

```diff
@@ -0,0 +1,184 @@
+import AnsiUp from 'ansi_up';
+import execa = require('execa');
+import { z } from 'zod';
+import * as Sentry from '@prairielearn/sentry';
+import { logger } from '@prairielearn/logger';
+import { loadSqlEquiv, queryAsync, queryValidatedOneRow } from '@prairielearn/postgres';
+
+import { chalk } from './chalk';
+import serverJobs = require('./server-jobs');
+import socketServer = require('./socket-server');
+
+const sql = loadSqlEquiv(__filename);
+
+interface CreateServerJobOptions {
+  courseId?: string;
+  courseInstanceId?: string;
+  courseRequestId?: string;
+  assessmentId?: string;
+  userId?: string;
+  authnUserId?: string;
+  type: string;
+  description: string;
+}
+
+interface ServerJobExecOptions {
+  cwd?: string;
```

Comment:
This should probably be a mandatory property - I can't imagine we'd ever want to just run commands in the working directory (the default)?

Reply:
I can't think of any reason that we wouldn't specify the working directory explicitly so I'm on board with making this argument mandatory.

Reply:
When trying to make this change, TypeScript pointed out a case where we don't _technically_ need an explicit directory: `git clone`.

Reply:
Ha, nice.

Reply:
I wouldn't object to forcing us to use `/` in this case, just for the certainty that it provides.

Reply:
Sounds reasonable. I made that change in 35f7710.